File: ui/src/api/type/message.ts
export type ApiMessage = {
  username: string
  timestamp: string
  title: string
  subtitle?: string
  body?: string
  id?: string
}

// example:
// { "username": "Test", "timestamp": "a", "title": "Test title", "subtitle": "Test subtitle", "body": "Test body.", "id": "1234" }

File: core/internal/domain/message.go
package domain

type Message struct {
	Username  string `json:"username"`
	Timestamp string `json:"timestamp"`
	Title     string `json:"title"`
	Subtitle  string `json:"subtitle"`
	Body      string `json:"body"`
	Id        string `json:"id"`
}

func (m Message) Validate() bool {
	return len(m.Title) > 0 && len(m.Username) > 0 && len(m.Timestamp) > 0
}
